### PR TITLE
feat: Artists result screen tab

### PR DESCRIPTION
**Pull request by rajsam003**
This PR resolves [GRO-1272] 

### Acceptance criteria

- [x] User sees artist rails of artists based on the artworks they saved 
- [x] User sees follow artist button 
- [x] User sees bio - limit after the first sentence on mobile
- [x] User sees an artwork rail with the first 25 artworks by the artist 

image

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [x] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[GRO-1272]: https://artsyproduct.atlassian.net/browse/GRO-1272?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by pvinis — DISMISSED** [review]
seems nice

File: src/app/Scenes/ArtQuiz/ArtQuizResults/ArtQuizResultsTabs/ArtQuizExploreArtist.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        content={artist?.biographyBlurb?.text!}
+        maxChars={textLimit}
+        textStyle="new"
+        textVariant="sm"
+        linkTextVariant="sm-display"
+      />
+      <Spacer m={1} />
+      {/* the negative margin here is for resetting padding of 20 that all the parent
+      components of this instance have and to avoid changing the component tree in multiple spots. */}
+      <Flex mx={-space("2")}>
```
Comment (MounirDhahri) [inline src/app/Scenes/ArtQuiz/ArtQuizResults/ArtQuizResultsTabs/ArtQuizExploreArtist.tsx:75]: **suggestion(non-blocker):** I would rather recommend to remove the parent from the `StickyTabFlatlist` as we do in [here](https://github.com/artsy/eigen/blob/0283f0d701675a1ba0bea7f8d9d2fd37d4bc3873/src/app/Scenes/Favorites/FavoriteCategories.tsx#L92) instead

File: src/app/Scenes/ArtQuiz/ArtQuizResults/ArtQuizResultsTabs/ArtQuizResultsTabs.tsx [hunk 2]
```diff
@@ -53,7 +54,8 @@ const artQuizResultsTabsFragment = graphql`
   fragment ArtQuizResultsTabs_me on Me {
     quiz {
       savedArtworks {
-        ...ArtQuizLikedArtworks_me
+        ...ArtQuizLikedArtworks_artworks
+        ...ArtQuizExploreArtists_artworks
       }
```
Comment (MounirDhahri) [inline src/app/Scenes/ArtQuiz/ArtQuizResults/ArtQuizResultsTabs/ArtQuizResultsTabs.tsx:59]: **issue:** As mentioned earlier, I would very strongly suggest against using this interface as it is now and instead use the paginated `SavedArtworksConnection` (or a similar interface) to return this list of artworks. with a p75 of 81 saved artworks per user, this is going to be a heavy a query that can explode quickly

File: src/app/Scenes/ArtQuiz/ArtQuizResults/ArtQuizResultsTabs/ArtQuizExploreArtist.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    artQuizExploreArtistFragment,
+    artistData
+  )
+
+  const artworks = extractNodes(artist?.artworksConnection)
+
+  const [followOrUnfollowArtist] =
+    useMutation<ArtQuizExploreArtistFollowArtistMutation>(FollowArtistMutation)
+
+  const handleFollowChange = (artist: ArtQuizExploreArtist_artist$data) => {
```
Comment (MounirDhahri) [inline src/app/Scenes/ArtQuiz/ArtQuizResults/ArtQuizResultsTabs/ArtQuizExploreArtist.tsx:32]: **suggestion:** I would suggest debouncing this call

File: src/app/Scenes/ArtQuiz/ArtQuizResults/ArtQuizResultsTabs/ArtQuizResultsTabs.tsx (hunk 2 above)
Comment (rajsam003) [reply src/app/Scenes/ArtQuiz/ArtQuizResults/ArtQuizResultsTabs/ArtQuizResultsTabs.tsx:59]: Somehow I missed using pagination. I will make that change. Thanks

**Comment by MounirDhahri** [comment]
@rajsam003 given the performance issues that we spoke about in Slack, I suggest some changes to make the UX smoother. Mainly getting `isFollowed` from the state and debouncing the calls. The changes are in the video

https://user-images.githubusercontent.com/11945712/216397300-21d91aa1-77ac-4185-98b4-5345f435de25.mov

**Review by TMcMeans — APPROVED** [review]

File: src/app/Scenes/ArtQuiz/ArtQuizResults/ArtQuizResultsTabs/ArtQuizResultsTabs.tsx (hunk 2 above)
Comment (laurabeth) [reply src/app/Scenes/ArtQuiz/ArtQuizResults/ArtQuizResultsTabs/ArtQuizResultsTabs.tsx:59]: If I'm understanding the above correctly, I don't think the concern is necessary.

This `savedArtworks` endpoint is customized for the art quiz. It fetches the internal id of each quiz artwork, and then fetches only the saved works among that subset of 16. It won't ever return all the saved works in a users collection. It also doesn't use a paginated resolver.

**Review by laurabeth — DISMISSED** [review]
looks good! :shipit: will try to rebuild from failed.

File: src/app/Scenes/ArtQuiz/ArtQuizArtworks.tsx [hunk 4]
```diff
@@ -179,16 +179,14 @@ export const ArtQuizArtworks = () => {
...
-            </Touchable>
-          </Flex>
+        <Flex flexDirection="row" justifyContent="center" mb={6}>
+          <Touchable onPress={() => handleNext("Dislike")} style={{ marginHorizontal: 40 }}>
+            <CloseIcon height={40} width={50} />
+          </Touchable>
+          <Spacer y="2" />
+          <Touchable onPress={() => handleNext("Like")} style={{ marginHorizontal: 40 }}>
+            <HeartIcon height={40} width={50} />
+          </Touchable>
```
Comment (rajsam003) [inline src/app/Scenes/ArtQuiz/ArtQuizArtworks.tsx:189]: Updating this as it was messed up after rebasing

File: src/app/Scenes/ArtQuiz/ArtQuizResults/ArtQuizResultsTabs/ArtQuizExploreArtist.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    artQuizExploreArtistFragment,
+    artistData
+  )
+
+  const artworks = extractNodes(artist?.artworksConnection)
+
+  const [followOrUnfollowArtist] =
+    useMutation<ArtQuizExploreArtistFollowArtistMutation>(FollowArtistMutation)
+
+  const handleFollowChange = debounce((artist: ArtQuizExploreArtist_artist$data) => {
```
Comment (rajsam003) [inline src/app/Scenes/ArtQuiz/ArtQuizResults/ArtQuizResultsTabs/ArtQuizExploreArtist.tsx:33]: @MounirDhahri 👆

File: src/app/Scenes/ArtQuiz/ArtQuizResults/ArtQuizResultsTabs/ArtQuizResultsTabs.tsx (hunk 2 above)
Comment (MounirDhahri) [reply src/app/Scenes/ArtQuiz/ArtQuizResults/ArtQuizResultsTabs/ArtQuizResultsTabs.tsx:59]: sounds good, thanks for the explanation. I thought it returns everything

**Review by MounirDhahri — APPROVED** [review]
LGTM thanks for addressing the comments Sam

**Pull request merged**